packages/@lwc/engine-core/src/framework/base-lightning-element.ts

```diff
@@ -175,12 +175,6 @@ function BaseLightningElementConstructor(this: LightningElement): LightningEleme
     if (isNull(vmBeingConstructed)) {
         throw new ReferenceError('Illegal constructor');
     }
-    if (process.env.NODE_ENV !== 'production') {
```

Comment:
This check is not needed after the spring cleaning.

Reply:
PM, can you please elaborate how `elm` is guaranteed to be an HTMLElement instance?
It wasn't clear from https://github.com/salesforce/lwc/pull/1893/files how it is guaranteed.

Reply:
There is no guaranty at the type level. We have struggled since the beginning of the SSR work to find the right balance with typings. We can't check in the engine if a given element is an instance of a give DOM constructor since it couple the engine with the DOM. I am open to any idea on how to work around this.

Reply:
I guess we could inject a validation function via the renderer or a similar mechanism?